src/probnum/filtsmooth/_gaussfiltsmooth/_kalman.py

```diff
@@ -231,121 +276,60 @@ def filtered_states_generator(
         --------
         RegressionProblem: a regression problem data class
         """
+
+        # Process inputs
         dataset, times = regression_problem.observations, regression_problem.locations
+        if not isinstance(measurement_model, abc.Iterable):
+            measurement_model = itertools.repeat(measurement_model, len(times))
 
-        _linearise_update_at = (
-            None if _previous_posterior is None else _previous_posterior(times[0])
-        )
+        # It is not clear at the moment how to implement this cleanly.
+        if not np.all(np.diff(times) > 0):
+            raise ValueError(
+                "Gaussian filtering cannot handle repeating time points currently."
+            )
```

Comment:
This also checks whether the time points are sorted, the error message, however, only reports that the locations are not _strictly_ sorted. Should we clarify that? Would have to be sth. like

```python
if not np.all(np.diff(times) >= 0):
   # either ...
   raise ValueError(
       "Locations have to be sorted"
   )
   #or
   times = np.sort(times)

if not np.all(np.diff(times) > 0):
       raise ValueError(
           "Gaussian filtering cannot handle repeating time points currently."
       )
```

Reply:
I see what you mean, good point. However, for simplicity reasons, I would rather change the current error message into "Gaussian filtering expects sorted, non-repeating time points." Otherwise that would be a lot of code for a very specific corner case?!

Reply:
Yep, sounds good.